kubernetes/linux/setup.sh

```diff
@@ -33,9 +33,9 @@ sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
 
 #install oneagent - Official bits (3/14/2022)
 if [ "${ARCH}" != "arm64" ]; then
-    wget "https://github.com/microsoft/Docker-Provider/releases/download/1.17.0/azure-mdsd_1.17.0-build.master.354_x86_64.deb" -O azure-mdsd.deb
+    wget "https://github.com/microsoft/Docker-Provider/releases/download/official%2Fmdsd%2F1.23.1/azure-mdsd_1.23.1-build.master.451_x86_64.deb" -O azure-mdsd.deb
```

Comment:
did we get the signoff AMA team on this build for both AMD64 and Arm64? and is perf data looks good?